compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala

```diff
@@ -2118,21 +2118,19 @@ object messages {
     override def explanation: String = ""
   }
 
-  case class NotAnExtractor(tree: tpd.Tree)(implicit ctx: Context) extends Message(NotAnExtractorID) {
-    override def msg: String = hl"""${tree.show} cannot be used as an extractor in a pattern because it is not implemented"""
+  case class NotAnExtractor(tree: Trees.Tree[Untyped])(implicit ctx: Context) extends Message(NotAnExtractorID) {
+    override def msg: String = hl"${tree.show} cannot be used as an extractor in a pattern because it lacks an unapply or unapplySeq method"
     override def kind: String = "Syntax"
     override def explanation: String =
-      hl"""| ${tree.show} is not a valid extractor for use in pattern matching,
+      hl"""| $tree is not a valid extractor for use in pattern matching,
            | this happens when the method unapply or unapplySeq it is not declared
            | on it's object companion.
```

Comment:
I would remove the first sentence of the explanation. When the error is reported with `-explain`, we print both `msg` and `explanation`. So the explanation should give additional information and not rephrase `msg`